File: wrappers/csharp/huestream_csharp_integrationtests/messages/FeedbackMessageObserver.cs
using System;
using System.Collections.Generic;
using System.Threading;
using huestream;
namespace huestream_tests
{
    public class FeedbackMessageObserver : FeedbackMessageHandler
    {
        public override void NewFeedbackMessage(FeedbackMessage message)
        {
            DisplayMessage(message);

            if (_isListening && message.GetId() == _idToListen)
            {
                _listeningEventHandle.Set();
            }
        }

        public void StartListening(FeedbackMessage.Id id)
        {
            if (_isListening)
                return;

            _idToListen = id;
            _isListening = true;
        }

        public void SetEventWaitHandle(EventWaitHandle handle)
        {
            _listeningEventHandle = handle;
        }

        private EventWaitHandle _listeningEventHandle;

        private bool _isListening = false;
        private FeedbackMessage.Id _idToListen;
    }
}

File: wrappers/java/android_integration_tests/app/src/androidTest/java/com/lighting/huestream/integrationtests/FeedbackMessageObserver.java
package com.lighting.huestream.integrationtests;

import com.lighting.huestream.FeedbackMessage;

public class FeedbackMessageObserver extends FeedBackHandler {
    @Override
    public synchronized void NewFeedbackMessage(FeedbackMessage message) {
        super.NewFeedbackMessage(message);

        if (_isListening && message.GetId().equals(_idToListen)) {
            _wasReceived = true;
            this.notify();
        }
    }

    public void StartListening(FeedbackMessage.Id id) {
        if (!_isListening) {
            _idToListen = id;
            _isListening = true;
        }
    }

    public boolean wasReceived() {
        return _wasReceived;
    }

    private boolean _isListening = false;
    private boolean _wasReceived = false;
    private FeedbackMessage.Id _idToListen;
}
